test/e2e/specs/editor/plugins/meta-boxes.spec.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Meta boxes', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-plugin-meta-box' );
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin( 'gutenberg-test-plugin-meta-box' );
+	} );
+
+	test( 'Should save the post', async ( { editor, page } ) => {
+		const saveDraft = page
+			.getByRole( 'region', { name: 'Editor top bar' } )
+			.getByRole( 'button', { name: 'Save draft' } );
+
+		// Save should not be an option for new empty post.
+		await expect( saveDraft ).toBeDisabled();
+
+		// Add title to enable valid non-empty post save.
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'Hello Meta' );
+
+		await expect( saveDraft ).toBeEnabled();
+
+		await editor.saveDraft();
+
+		// After saving, affirm that the button returns to Save Draft.
+		await expect( saveDraft ).toBeEnabled();
+	} );
+
+	test( 'Should render dynamic blocks when the meta box uses the excerpt for front end rendering', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		// Publish a post so there's something for the latest posts dynamic block to render.
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'A published post' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Hello there!' );
+		await editor.publishPost();
+
+		// Publish a post with the latest posts dynamic block.
+		await admin.createNewPost();
+		await page
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.fill( 'Dynamic block test' );
+		await editor.insertBlock( { name: 'core/latest-posts' } );
+
+		const postId = await editor.publishPost();
+		await page.goto( `/?p=${ postId }` );
+
+		await expect(
+			page.locator( '.wp-block-latest-posts > li' )
+		).toContainText( [ 'A published post', 'Dynamic block test' ] );
```

Comment:
Hey, I noticed this causing a couple flaky tests today and grabbed a screenshot:

![image](https://github.com/WordPress/gutenberg/assets/6265975/e1490b63-0fd4-4df5-af7e-dd0b97471e82)

And here's a link to the test results: https://github.com/WordPress/gutenberg/actions/runs/6846766063/job/18613885820

It looks like the order of the text can be reversed in the post, but `.toContainText` enforces the exact order of the input array

Reply:
Thank you, @noahtallen! I'll have look shortly.